nomad/job_endpoint.go

```diff
@@ -328,52 +349,51 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 			return err
 		}
 
+		submittedEval = true
+
 		// Populate the reply with job information
 		reply.JobModifyIndex = index
+		reply.Index = index
+
+		if args.Eval != nil {
+			reply.EvalCreateIndex = index
+		}
+
 	} else {
 		reply.JobModifyIndex = existingJob.JobModifyIndex
 	}
 
 	// used for multiregion start
 	args.Job.JobModifyIndex = reply.JobModifyIndex
 
-	// If the job is periodic or parameterized, we don't create an eval.
-	if args.Job.IsPeriodic() || args.Job.IsParameterized() {
+	if args.Eval == nil {
 		return nil
 	}
 
-	// Create a new evaluation
-	now := time.Now().UTC().UnixNano()
-	eval := &structs.Evaluation{
-		ID:             uuid.Generate(),
-		Namespace:      args.RequestNamespace(),
-		Priority:       args.Job.Priority,
-		Type:           args.Job.Type,
-		TriggeredBy:    structs.EvalTriggerJobRegister,
-		JobID:          args.Job.ID,
-		JobModifyIndex: reply.JobModifyIndex,
-		Status:         structs.EvalStatusPending,
-		CreateTime:     now,
-		ModifyTime:     now,
-	}
-	update := &structs.EvalUpdateRequest{
-		Evals:        []*structs.Evaluation{eval},
-		WriteRequest: structs.WriteRequest{Region: args.Region},
-	}
+	// COMPAT(1.1.0): Remove the ServerMeetMinimumVersion check.
+	// 0.12.1 introduced atomic eval job registration
+	if args.Eval != nil &&
```

Comment:
I think this `args.Eval != nil` is always true; we're checking `args.Eval == nil` right above it.

Reply:
Actually, do we need to remove the returning early above for multi region deployment?  Returning early means that periodic/dispatch jobs will not be handled by `multiregionStart`?  This PR doesn't change the behavior, but just noticed the MRD call.

Reply:
Periodic and dispatch jobs get kicked off with their normal dispatch mechanisms in MRD. We special case them in `schedule/reconcile.go`, rather than running them through the MRD in deploymentwatcher.

Reply:
In that case, I'm tempted to remove the early return and somehow make it more `multiregionStart` that doesn't apply to paramterized/periodic jobs.

Reply:
Not a bad idea. There's a big blog of `return nil` at the top of the ENT functionality where we return early when we don't need it.

Reply:
Actually, I'll follow up in another PR - I'm a bit confused about the interaction and would love to do more testing.